packages/vite/src/node/cli.ts

```diff
@@ -263,31 +278,58 @@ cli
     `[boolean] force empty outDir when it's outside of root`,
   )
   .option('-w, --watch', `[boolean] rebuilds when modules have changed on disk`)
-  .action(async (root: string, options: BuildOptions & GlobalCLIOptions) => {
-    filterDuplicateOptions(options)
-    const { build } = await import('./build')
-    const buildOptions: BuildOptions = cleanOptions(options)
+  .option('--environment [name]', `[string] build a single environment`)
+  .option('--app', `[boolean] same as builder.entireApp`)
```

Comment:
`/guide/cli.md` needs to be updated later.

Reply:
Added it as experimental to the cli docs.

Reply:
I noticed that `--environment` wasn't added to the docs and looks like it's not used in the code. Should line 281 be removed?

Reply:
Removed it at [refactor: vite build cli implementation](https://github.com/vitejs/vite/pull/16471/commits/74dc901e146fff9e763bb2b0508fc8767f1bee61), and cleaned up the CLI implementation to avoid duplicating the logic we had in the `build()` function.

About `--environment`. At the beginning we wanted to follow the way the `--ssr` flag works, generalizing it to any environment name. But we decided later on that building only part of the app was not worth exposing in the CLI as it makes things more complex once there is a `buildApp()` config defined that may do other tasks as part of building the environments, or may require one environment to be build before the other, or sharing data. So `--environment` in general may fail in these applications. For now, we aren't going to expose it to check if building the whole application is enough.